### PR TITLE
some minor fixes & additions:

### PR DESCRIPTION
**Pull request by mnunberg**
- add a setup/teardown feature
- better cleanup for `--exit-on-failure`
- allow for a `message` argument in most `assert()` functions

### CODE REVIEW

File: RLTest/__main__.py [hunk 1]
```diff
@@ -404,28 +405,39 @@ def _runTest(self, test, numberOfAssertionFailed=0, prefix=''):
                 return 0
 
             fn = lambda: test.target(env)
+            before = (lambda: before(env)) if before else None
```
Comment (MeirShpilraien) [inline RLTest/__main__.py]: why we need to set the before in lambda? why not just calling before directly?

Comment (mnunberg) [reply RLTest/__main__.py]: it wraps the lambda so that we don't need to know if it needs the argument

Comment (MeirShpilraien) [reply RLTest/__main__.py]: Why not just passing the argument anyway ?

**Comment by mnunberg** [comment]
I've updated the PR per the comments

**Pull request merged**